src/lib/fetch-document-list.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-console */
+const { info } = require('../lib/log');
+const fs = require('../lib/sync-fs');
+const DOC_TYPES = ['district_hospital', 'health_center', 'clinic', 'person', 'user', 'user-settings', 'contact'];
+
+const jsonDocPath = (directoryPath, docID) => `${directoryPath}/${docID}.doc.json`;
+
+const fetchDocumentListFromDB = async (db, ids) => {
+  info('Downloading doc(s)...');
+  const documentDocs = await db.allDocs({
+    keys: ids,
+    include_docs: true,
+  });
+
+  const errors =  { invalidDocType: [], docNotFound: [] };
+
+  documentDocs.rows.forEach(row => {
+    if (!row.doc) {
+        errors.docNotFound.push(`Document with id '${row.key}' could not be found.`);
+        return;
+    }
+
+    if (!DOC_TYPES.includes(row.doc.type)) {
+      errors.invalidDocType.push(`Document with id ${row.key} of type ${row.doc.type} cannot be edited`);
+    }
+  });
```

Comment:
Please move this to its own function. it will make the overall function a lot easier to read. As it is now, this function has 3 parts: load data, check for bad data, and do the normal case (transforming the data). The bulk of the lines though are taken up by error checking logic. Moving that to a separate function will allow readers to stay in that easy mode of reading as enumerated above.

I'd recommend a new function something like `findErrors`, which just does this same stuff and returns back the object like on line 15.